### PR TITLE
Remove private constructors to fix the issue in other db connectors

### PR DESCRIPTION
**Pull request by lahirusamith**
## Purpose
$title

Fixes:

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Review by daneshk — APPROVED** [review]

File: sql-native/src/main/java/org/ballerinalang/sql/utils/RecordIteratorUtils.java [hunk 1]
```diff
@@ -59,8 +59,6 @@
  * @since 1.2.0
  */
 public class RecordIteratorUtils {
-    private RecordIteratorUtils() {
```
Comment (niveathika) [inline sql-native/src/main/java/org/ballerinalang/sql/utils/RecordIteratorUtils.java:62]: Only this constructor is exposed, isn't it? Let's make this constructor public and keep the rest. I will look into the code quality aspects of this

Comment (lahirusamith) [reply sql-native/src/main/java/org/ballerinalang/sql/utils/RecordIteratorUtils.java:62]: `DefaultResultParameterProcessor` this class also exposed

Comment (niveathika) [reply sql-native/src/main/java/org/ballerinalang/sql/utils/RecordIteratorUtils.java:62]: Ack.

Comment (kalaiyarasiganeshalingam) [reply sql-native/src/main/java/org/ballerinalang/sql/utils/RecordIteratorUtils.java:62]: Yes, We can change it to `public`

**Review by niveathika — APPROVED** [review]

**Pull request merged**